### PR TITLE
[NFTs] Implement missed methods to set the attributes from other pallets

### PR DESCRIPTION
**Pull request by jsidorenko**
In this PR, I added implementations for the `set_attribute`/`set_typed_attribute` + `set_collection_attribute`/`set_typed_collection_attribute` methods so that it would be possible to set attributes from the other pallets into a special `Pallet` namespace.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2153524

**Review by gilescope — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2153614

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2153797

**Review by ruseinov — APPROVED** [review]

File: frame/support/src/lib.rs [hunk 1]
```diff
@@ -127,7 +127,7 @@ pub const LOG_TARGET: &str = "runtime::frame-support";
 pub enum Never {}
 
 /// A pallet identifier. These are per pallet and should be stored in a registry somewhere.
-#[derive(Clone, Copy, Debug, Eq, PartialEq, Encode, Decode, TypeInfo, MaxEncodedLen)]
+#[derive(Clone, Copy, Eq, PartialEq, Encode, Decode, TypeInfo)]
 pub struct PalletId(pub [u8; 8]);
```
Comment (muharem) [inline frame/support/src/lib.rs:131]: this is a public api of frame.
why you removing MaxEncodedLen derive?
Your assumption that it is not used by anyone and anything expect your use case?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2155016

File: frame/support/src/lib.rs (hunk 1 above)
Comment (jsidorenko) [reply frame/support/src/lib.rs:131]: Sorry, I needed to set the context. I've added this in the prev PR where I needed the PalletId, but now we decided to remove it from the namespace `Pallet(PalletId)` => `Pallet`. That's why I simply reverting the prev changes

**Review by muharem — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2155272

**Pull request merged**